spec/requests/api/v2/site_statuses_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'rails_helper'
+
+describe 'Site Helpers API V2' do
+  let(:user) { create(:user).tap { |u| organisation.users << u } }
+  let(:organisation) { site_status.course.provider.organisations.first }
+  let(:payload) { { email: user.email } }
+  let(:token) do
+    JWT.encode payload,
+                Settings.authentication.secret,
+                Settings.authentication.algorithm
+  end
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+  let(:site_status) { create :site_status }
+  let(:params)      { {} }
+  let(:perform_request) do
+    patch(
+      api_v2_site_status_path(site_status),
+      headers: { 'HTTP_AUTHORIZATION' => credentials },
+      params: params
+    )
+  end
+
+  subject { response }
+
+  describe 'PATCH update' do
+    context 'when unauthenticated' do
+      let(:payload) { { email: 'foo@bar' } }
+
+      before do
+        perform_request
+      end
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when unauthorised' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload) { { email: unauthorised_user.email } }
+
+      it 'raises an error' do
+        expect { perform_request }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when authorised' do
+      let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+      let(:params) do
+        {
+          _jsonapi: jsonapi_renderer.render(
+            site_status,
+            class: {
+              SiteStatus: API::V2::SerializableSiteStatus
+            }
+          )
+        }
+      end
+      let(:site_status_params)         { params.dig :_jsonapi, :data, :attributes }
+      let(:applications_accepted_from) { '2019-01-01 00:00:00' }
+      let(:publish)                    { 'published' }
+      let(:status)                     { 'discontinued' }
+      let(:vac_status)                 { 'no_vacancies' }
+      let(:json_data)                  { JSON.parse(response.body)['data'] }
+
+      before do
+        site_status_params[:applications_accepted_from] = applications_accepted_from
+        site_status_params[:publish]                    = publish
+        site_status_params[:status]                     = status
+        site_status_params[:vac_status]                 = vac_status
```

Comment:
For this iteration, let's use `merge!` here to make it a little clearer that we're updating a hash in-place.